internal/exec/exec.go

```diff
@@ -51,7 +51,9 @@ func (r *Request) Execute(ctx context.Context, s *resolvable.Schema, op *query.O
 	}()
 
 	if err := ctx.Err(); err != nil {
-		return nil, []*errors.QueryError{errors.Errorf("%s", err)}
+		//If context has either been cancelled or timed out we still may want to return the features that have finished
+		// TODO properly attribute mark which features have timedout in the error field
+		return out.Bytes(), []*errors.QueryError{errors.Errorf("%s", err)}
```

Comment:
Is there a way to only add our new behaviour if the error was a context deadline timeout?

Reply:
Is it too brittle to switch on the error being "context deadline exceeded"?

Reply:
As of time of writing the only two errors that context.Error() will return are if the context deadline is exceeded or the context is canceled.

https://golang.org/pkg/context/#Context